src/services/msal.ts

```
@@ -1,22 +1,28 @@
+import {ConfidentialClientApplication, LogLevel} from '@azure/msal-node';
 
-const msalConfig: Configuration = {
-    auth: {
-        clientId: configuration.msal.applicationId,
-        authority: `https://login.microsoftonline.com/${configuration.msal.tenantId}`,
-        clientSecret: configuration.msal.secret,
+let _redirectUri: URL;
+let _msal: ConfidentialClientApplication
+export default {
+    get redirectUri() {
+        return _redirectUri ??= new URL('/.auth/login/callback', configuration.site);
```

Biome (1.9.4)

[error] The assignment should not be in an expression.

The use of assignments in expressions is confusing.
Expressions are often considered as side-effect free.

(lint/suspicious/noAssignInExpressions)